subprojects/docs/src/docs/userguide/groovy_plugin.adoc

```diff
@@ -222,34 +222,35 @@ The Groovy plugin adds a link:{groovyDslPath}/org.gradle.api.tasks.compile.Groov
 [[sec:groovy_compilation_avoidance]]
 === Compilation avoidance
 
-_Caveat: the Groovy compilation avoidance is an incubating feature since Gradle 5.6. There're known inaccuracies so please enable it at your own risk._
+_Caveat: Groovy compilation avoidance is an incubating feature since Gradle 5.6. There're known inaccuracies so please enable it at your own risk._
 
 To enable the incubating support for Groovy compilation avoidance, add an system property `org.gradle.groovy.compilation.avoidance` to your Gradle invocation, for example:
 
 ```
 ./gradlew assemble -Dorg.gradle.groovy.compilation.avoidance=true
 ```
 
-If a dependent project has changed in an https://en.wikipedia.org/wiki/Application_binary_interface[ABI]-compatible way (only its private API has changed),
-then Groovy compilation tasks are supposed to be up-to-date. This means that if project `A` depends on project `B` and a class in `B` is changed in an ABI-compatible way (typically, changing only the body of a method), then Gradle won't recompile `A`.
+Or, add the system property to your `gradle.properties` file:
+
+
+```
+# gradle.properties file
+org.gradle.jvmargs=<original arguments> -Dorg.gradle.groovy.compilation.avoidance=true
```

Comment:
Use `systemProp.org.gradle.groovy.compilation.avoidance=true` instead of changing `org.gradle.jvmargs`.